src/sched/basic.c

```diff
@@ -107,31 +106,25 @@ static void sched_run(ABT_sched sched)
     event_freq = p_data->event_freq;
     num_pools  = p_data->num_pools;
     pools      = p_data->pools;
+    ABT_unit unit = ABT_UNIT_NULL;
```

Comment:
Please declare it either at the beginning of the function or at the beginning of the `while` loop.